### PR TITLE
Upstream argocd components

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Upstream argocd components

## why
* Upstream

## references
* N/A

### CODE REVIEW

File: modules/eks/argocd/main.tf [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  kubernetes_namespace = var.kubernetes_namespace
+  sops_directory       = abspath("./resources")
+  sops_file_name       = "argocd-notifications-secret.sops.yaml"
+}
+
+module "argocd" {
+  source  = "cloudposse/helm-release/aws"
+  version = "0.3.0"
+
+  name                   = "" # avoids hitting length restrictions on IAM Role names
```
Comment (Gowiem) [inline modules/eks/argocd/main.tf:114]: @nitrocode what does your argocd chart install end up getting named? Due to the code [here](https://github.com/cloudposse/terraform-aws-helm-release/blob/master/main.tf#L41), we get funky names for our argocd installation like `argo-cd-argocd-server`, which I am not a fan of. What do your K8s resource names look like with this chart and have you found a way around that name duplication?

Comment (nitrocode) [reply modules/eks/argocd/main.tf:114]: Ah that's a good question. Im unsure tbh

Comment (Gowiem) [reply modules/eks/argocd/main.tf:114]: @nitrocode it's that you folks override `fullnameOverride` for all of your chart deployments I believe. This does away with whatever is passed in as the name 👍

**Comment by osterman** [comment]
Superceeded by #560

**Pull request closed without merging**